src/app/services/dsm.service.ts

```diff
@@ -548,12 +551,13 @@ export class DSMService {
     map.push( {name: "userId", value: this.role.userID()} );
     return this.http.post(url, null, this.buildQueryHeader(map)).pipe(
       catchError(this.handleError)
-    );
-  }
-
-  public singleKitLabel( kitJson: string ): Observable<any> {
+      );
+    }
+    
+    public singleKitLabel(realm: string, kitJson: string ): Observable<any> {
     let url = this.baseUrl + DSMService.UI + "kitLabel";
     let map: { name: string, value: any }[] = [];
+    map.push( {name: DSMService.REALM, value: realm} );
```

Comment:
maybe doing it here same as for `setKitSentRequest`: map.push( {name: DSMService.REALM, value: localStorage.getItem(ComponentService.MENU_SELECTED_REALM)} );
and not adding it to the method?